In [2]:
import pandas as pd

df = pd.read_csv("bicicletas_compartidas_bogota.csv")


In [2]:

df.head()

,viaje_id,inicio_viaje,estacion_origen,localidad_origen,estacion_destino,localidad_destino,duracion_min,distancia_km,tipo_bicicleta,tipo_usuario,clima,temperatura_c,precipitacion_mm,fin_de_semana
0,V000671,2026-04-22 08:35:33,Plaza de Bolívar,La Candelaria,Universidades Centro,La Candelaria,13.8,4.14,Eléctrica,Suscriptor,Nublado,17.2,0.0,No
1,V002336,2026-03-26 20:39:41,Plaza de Bolívar,La Candelaria,Universidades Centro,La Candelaria,10.8,2.47,Eléctrica,Ocasional,Lluvia ligera,15.2,0.9,No
2,V002504,2026-03-09 11:25:42,Parque de los Novios,Barrios Unidos,Hacienda Santa Bárbara,Usaquén,6.6,0.83,Eléctrica,Suscriptor,Nublado,17.1,0.0,No
3,V002587,2026-04-29 04:04:46,Parque de la 93,Chapinero,Parque de la 93,Chapinero,7.6,1.68,Mecánica,Suscriptor,Nublado,21.0,0.0,No
4,V000820,2026-05-26 03:55:26,Zona G,Chapinero,Parque Nacional,Santa Fe,16.6,2.89,Mecánica,Suscriptor,Lluvia,18.7,7.8,No


In [3]:
df.shape

(3214, 14)

In [4]:
df.columns

Index(['viaje_id', 'inicio_viaje', 'estacion_origen', 'localidad_origen',
       'estacion_destino', 'localidad_destino', 'duracion_min', 'distancia_km',
       'tipo_bicicleta', 'tipo_usuario', 'clima', 'temperatura_c',
       'precipitacion_mm', 'fin_de_semana'],
      dtype='str')

In [5]:
df.describe()

,duracion_min,distancia_km,temperatura_c,precipitacion_mm
count,3214.000000,3162.000000,3169.000000,3214.000000
mean,14.231518,2.853033,16.647144,0.996235
std,8.670122,1.698718,2.356015,2.459092
min,0.500000,0.050000,8.000000,0.000000
25%,8.900000,1.710000,15.100000,0.000000
50%,12.500000,2.480000,16.600000,0.000000
75%,17.600000,3.580000,18.300000,0.150000
max,180.000000,31.200000,24.900000,12.000000


In [6]:
df.dtypes

viaje_id                 str
inicio_viaje             str
estacion_origen          str
localidad_origen         str
estacion_destino         str
localidad_destino        str
duracion_min         float64
distancia_km         float64
tipo_bicicleta           str
tipo_usuario             str
clima                    str
temperatura_c        float64
precipitacion_mm     float64
fin_de_semana            str
dtype: object

In [14]:
df[["distancia_km","duracion_min"]].corr()

,distancia_km,duracion_min
distancia_km,1.000000,0.721721
duracion_min,0.721721,1.000000


In [41]:
d = df.sort_values("distancia_km",ascending=False)
d.head()

,viaje_id,inicio_viaje,estacion_origen,localidad_origen,estacion_destino,localidad_destino,duracion_min,distancia_km,tipo_bicicleta,tipo_usuario,clima,temperatura_c,precipitacion_mm,fin_de_semana
2089,V000844,2026-03-03 20:53:19,Parque Nacional,Santa Fe,Las Nieves,Santa Fe,8.9,31.20,Mecánica,Suscriptor,Despejado,15.9,0.0,No
1052,V002209,2026-05-11 09:21:29,Parque de los Hippies,Chapinero,Parque de los Novios,Barrios Unidos,22.3,27.50,electrica,Suscriptor,Lluvia,16.2,11.6,No
404,V001732,2026-03-31 19:36:50,Calle 116,Usaquén,Galerías,Teusaquillo,58.2,13.70,Mecánica,Ocasional,Despejado,16.4,0.0,No
1952,V001332,2026-05-19 04:59:43,Las Aguas,La Candelaria,NaN,Chapinero,61.9,10.73,Mecánica,Suscriptor,Despejado,15.9,0.0,No
2300,V000108,2026-03-24 19:24:17,Unicentro,Usaquén,Parkway,Teusaquillo,40.3,10.63,electrica,Suscriptor,Despejado,13.7,0.0,No



a) p = 0.7217

b) Podemos afirmar que exiate una fuerte relación lineal poositiva entre ambas variables, lo que quiere decir que cuando aumentan la distacia del recorrido tambien aumenta el tiempo de desplazamiento.

c ) No seria correcto afirmar esto, puesto que aunque hay una relacion lineal positiva muy fuerte no significa que exista una razon de causalidad  y simplemente puede ser una relacion espuria.

d) Una correlacion alta no implica que necesariamente una mayor distancia tenga un mayor tiempo como por ejemplo los viajes con el id
V001732 que tiene una distancia de 31.20 km tiene una duración de 8,9 min y en cambio el viaje v002209 que tiene una distancia de 27,5 km tiene un tiempo de 22,3 min, mucho mas alto que el anterior teniendo una distancia menor


In [3]:
df[["distancia_km","duracion_min"]].describe()

,distancia_km,duracion_min
count,3162.000000,3214.000000
mean,2.853033,14.231518
std,1.698718,8.670122
min,0.050000,0.500000
25%,1.710000,8.900000
50%,2.480000,12.500000
75%,3.580000,17.600000
max,31.200000,180.000000


In [6]:
IQRDIS = 3.580000 - 1.710000
IQRDUR = 17.600000 - 8.900000

In [3]:
df["tipo_bicicleta"] = df["tipo_bicicleta"].str.replace("Eléctrica", "electrica")

In [4]:
df.groupby("tipo_bicicleta")["distancia_km"].mean()

tipo_bicicleta
Mecánica     2.890151
electrica    2.812424
Name: distancia_km, dtype: float64

In [5]:
df.groupby("tipo_bicicleta")["duracion_min"].mean()

tipo_bicicleta
Mecánica     15.382002
electrica    12.974674
Name: duracion_min, dtype: float64

In [6]:
df.groupby("tipo_bicicleta")["duracion_min"].median()

tipo_bicicleta
Mecánica     13.55
electrica    11.50
Name: duracion_min, dtype: float64

In [13]:
df.groupby("tipo_bicicleta").size()

tipo_bicicleta
Mecánica     1678
electrica    1536
dtype: int64

| Tipo de bicicleta | Número de viajes | Distancia promedio (km) | Duración promedio (min) | Duración mediana (min) |
|-------------------|------------------|--------------------------|---------------------------|---------------------------|
| Mecánica          | 1678             | 2.890151                 | 15.382002                  | 13.55                      |
| Eléctrica         | 1536             | 2.812424                 | 12.974674                  | 11.50                      |

4) Si por lo que vemos en la tabla las electricas tienen una duracion menor promedio teniendo una distancia muy similar (2,89km vs 2,81km), y en la mediana tenemos la misma tendencia con electricas que tienen menor duración.

In [32]:
df["clima"]= df["clima"].str.replace("Lluvia ligera","lluvia ligera")

In [33]:
df.groupby("clima").size()

clima
Despejado        1337
Lluvia            358
Nublado          1059
lluvia ligera     460
dtype: int64

In [35]:
df.groupby("clima")["distancia_km"].mean()

clima
Despejado        2.854924
Lluvia           2.940600
Nublado          2.804957
lluvia ligera    2.889978
Name: distancia_km, dtype: float64

In [21]:
df.groupby("clima")["duracion_min"].mean()

clima
Despejado        14.272251
Lluvia           14.310615
Nublado          14.037205
lluvia ligera    14.498913
Name: duracion_min, dtype: float64

In [22]:
df.groupby("clima")["duracion_min"].median()

clima
Despejado        12.50
Lluvia           12.75
Nublado          12.30
lluvia ligera    12.90
Name: duracion_min, dtype: float64

In [38]:
d = df.query("distancia_km >= 2 and distancia_km <= 4")

In [30]:
d.groupby("clima").size()

clima
Despejado        607
Lluvia           162
Nublado          490
lluvia ligera    212
dtype: int64

In [34]:
d.groupby("clima")["distancia_km"].mean()

clima
Despejado        2.826161
Lluvia           2.866914
Nublado          2.813286
lluvia ligera    2.883160
Name: distancia_km, dtype: float64

In [27]:
d.groupby("clima")["duracion_min"].mean()

clima
Despejado        14.506590
Lluvia           14.465432
Nublado          14.524490
lluvia ligera    14.433491
Name: duracion_min, dtype: float64

In [28]:
d.groupby("clima")["duracion_min"].median()

clima
Despejado        13.90
Lluvia           13.85
Nublado          13.65
lluvia ligera    13.75
Name: duracion_min, dtype: float64

5.
a)
| Clima            | Número de viajes | Distancia promedio (km) | Duración promedio (min) | Duración mediana (min) |
|-------------------|-------------------|---------------------------|---------------------------|---------------------------|
| Despejado         | 1337              | 2.854924                  | 14.272251                  | 12.50                      |
| Lluvia            | 358               | 2.940600                  | 14.310615                  | 12.75                      |
| Lluvia ligera     | 460               | 2.889978                  | 14.498913                  | 12.90                      |
| Nublado           | 1059              | 2.804957                  | 14.037205                  | 12.30                      |

b)
| Clima            | Número de viajes | Distancia promedio (km) | Duración promedio (min) | Duración mediana (min) |
|-------------------|-------------------|---------------------------|---------------------------|---------------------------|
| Despejado         | 607               | 2.826161                  | 14.506590                  | 13.90                      |
| Lluvia            | 162               | 2.866914                  | 14.465432                  | 13.85                      |
| Lluvia ligera     | 212               | 2.883160                  | 14.433491                  | 13.75                      |
| Nublado           | 490               | 2.813286                  | 14.524490                  | 13.65                      |

c)
-Las variables no cambian mucho, la que  cambia es la cantidad de viajes cuando cambia el clima. Cuando es un clima lluvioso podemos apreciar que la cantidad de viajes disminuye considerablemente.
-
-Podría ser una variable de confusión, porque a mayor distancia el viaje dura más y quizás está relacionada con el clima. Pero cuando controlo por distancia, las diferencias entre climas casi no cambian, así que su efecto parece ser mínimo.
-